hooks/useRailgunTx.tsx

```diff
@@ -0,0 +1,188 @@
+import { useToken } from "@/contexts/TokenContext";
+import useShieldPrivateKey from "@/hooks/useShieldPrivateKey";
+import {
+  gasEstimateForShield,
+  gasEstimateForShieldBaseToken,
+  populateShield,
+  populateShieldBaseToken,
+} from "@railgun-community/quickstart";
+import {
+  RailgunERC20AmountRecipient,
+  RailgunERC20Amount,
+  NetworkName,
+  TransactionGasDetailsSerialized,
+  EVMGasType,
+  NETWORK_CONFIG,
+  deserializeTransaction,
+} from "@railgun-community/shared-models";
+import { ethers } from "ethers";
+import { getAddress, parseUnits } from "ethers/lib/utils.js";
+import { useAccount, useSigner } from "wagmi";
+import { useProvider } from "wagmi";
+
+const useRailgunTx = () => {
+  const { data: signer } = useSigner();
+  const { address } = useAccount();
+  const { getShieldPrivateKey } = useShieldPrivateKey();
+  const network = NetworkName.EthereumGoerli;
+  const { weth } = useToken();
+  const provider = useProvider();
+
+  const shield = async (args: {
+    tokenAddress: string;
+    tokenAmount: string;
+    tokenDecimals: number;
+    recipient: string;
+  }) => {
+    if (args.tokenAddress === getAddress(`0x${"e".repeat(40)}`))
+      return shieldBaseToken(args);
+    return shieldToken(args);
+  };
+
+  const shieldBaseToken = async ({
+    tokenAmount,
+    tokenDecimals,
+    recipient,
+  }: {
+    tokenAddress: string;
+    tokenAmount: string;
+    tokenDecimals: number;
+    recipient: string;
+  }) => {
+    // Public wallet to shield from.
+    const fromWalletAddress = address as `0x{string}`;
+
+    // The shieldPrivateKey enables the sender to decrypt
+    // the receiver's address in the future.
+    const shieldPrivateKey = await getShieldPrivateKey();
+
+    const wrappedERC20Amount: RailgunERC20Amount = {
+      tokenAddress: weth!, // wETH
+      amountString: parseUnits(tokenAmount!, tokenDecimals).toHexString(), // hexadecimal amount
+    };
+
+    const { gasEstimateString, error: err } =
+      await gasEstimateForShieldBaseToken(
+        network,
+        recipient,
+        shieldPrivateKey,
+        wrappedERC20Amount,
+        fromWalletAddress
+      );
+    if (err) {
+      throw err;
+    }
+
+    const { maxFeePerGas, maxPriorityFeePerGas } = await provider.getFeeData();
+
+    const gasDetailsSerialized: TransactionGasDetailsSerialized = {
+      evmGasType: EVMGasType.Type2, // Depends on the chain (BNB uses type 0)
+      gasEstimateString: gasEstimateString!, // Output from gasEstimateForShieldBaseToken
+      maxFeePerGasString: maxFeePerGas!.toHexString(), // Current gas Max Fee
```

Comment:
Do we have an issue tracking all the changes we need when we convert from just Goerli support to supporting all chains? Seems the single-chain assumptions are hardcoded and spread across various locations, worried it may get a bit tedious to refactor. Probably fine though

Reply:
i wrote an issue #48 for the gas type

Reply:
Sounds great, thanks!